File: problems-extend-adapt/1901-2000/1991_array-balance-point/solution.go
func arrayBalancePoint(nums []int) int {
	// Single pass with a running left sum: an index is a middle index when
	// left == total - left - nums[i] (the right side's sum).
	total := 0
	for _, x := range nums {
		total += x
	}
	left := 0
	for i, x := range nums {
		if left == total-left-x {
			return i
		}
		left += x
	}
	return -1
}

File: problems-extend-adapt/1901-2000/1991_array-balance-point/solution.java
class Solution {

    public int arrayBalancePoint(int[] nums) {
        // Single pass with a running left sum: an index is a middle index
        // when left == total - left - nums[i] (the right side's sum).
        int total = 0;
        for (int x : nums) total += x;
        int left = 0;
        for (int i = 0; i < nums.length; ++i) {
            if (left == total - left - nums[i]) return i;
            left += nums[i];
        }
        return -1;
    }
}
